torch/distributed/elastic/timer/file_based_local_timer.py

```diff
@@ -209,117 +204,97 @@ def run_once(self) -> None:
             self._watchdog_thread = None
         else:
             log.info("No watchdog thread running, doing nothing")
-        if os.path.exists(self._file_path):
-            os.remove(self._file_path)
 
     def _watchdog_loop(self) -> None:
-        # Open the pipe in blocking mode blocks the server thread.
-        # This is fine for the following reasons:
-        #  1. No client case usually does not happen.
-        #  2. We are running the watchdog loop in a separate daemon
-        #     thread, which will not block the process to stop.
-        with open(self._file_path) as fd:
-            while not self._stop_signaled:
-                try:
-                    run_once = self._run_once
-                    self._run_watchdog(fd)
-                    if run_once:
-                        break
-                except Exception:
-                    log.exception("Error running watchdog")
-
-    def _run_watchdog(self, fd: io.TextIOWrapper) -> None:
-        timer_requests = self._get_requests(fd, self._max_interval)
+        while not self._stop_signaled:
+            try:
+                run_once = self._run_once
+                self._load_clients()
+                self._run_watchdog()
+                if run_once:
+                    break
+            except Exception:
+                log.exception("Error running watchdog")
+
+    def _load_clients(self) -> None:
+        if len(self._clients_metadata) == self._num_clients:
+            return
+
+        client_files = os.listdir(self._dir_path)
+        for file in client_files:
+            if file not in self._clients_metadata:
+                file_name = f"{self._dir_path}/{file}"
+                with open(file_name) as fd:
+                    request = json.loads(fd.readline())
+                    self._clients_metadata[file] = FileTimerRequest(
```

Comment:
does this need to be wrapped in a try-catch for partial reads/corrupted files? or do we want to fail in that case?